kqueen/auth.py

```diff
@@ -40,3 +42,40 @@ def identity(payload):
     except Exception:
         user = None
     return user
+
+
+def is_authorized(user, policy_value):
+    """
+    Evaluate if given user fulfills requirements of the given
+    policy_value.
+
+    Example:
+        >>> user.get_dict()
+        >>> {'username': 'jsmith', ..., 'role': 'member'}
+        >>> is_authorized(user, "all")
+        True
+        >>> is_authorized(user, ['admin'])
+        False
+
+    Args:
+        user (dict or User): User data
+        policy_value (string or list): Either "all" or list of allowed roles
+
+    Returns:
+        bool: authorized or not
+    """
+    if isinstance(user, User):
+        role = user.get_dict()['role']
+    elif isinstance(user, dict):
```

Comment:
Why do we need to support passing user as `dict`. I'd prefer to pass `User` object everywhere.

Reply:
Because what we have at hand in API is not actually User object. I tried it and for whatever reason current_identity is not actually instance of User (even though it definitely looks that way), its some kind of werkzeug proxy instance. So I have to pass current_identity.get_dict() as argument for this method in generic_views.

Reply:
Thanks, got it.